### PR TITLE
basedpyright 1.15.2 (new formula)

### PR DESCRIPTION
**Pull request by nikaro**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

basepyright is a fork/repackaging of pyright (with some more features).

### CODE REVIEW

**Comment by branchvincent** [comment]
Thanks for the PR, the python package is just a wrapper around the node package so we should use that instead. You can copy [`pyright` ](https://github.com/Homebrew/homebrew-core/blob/master/Formula/p/pyright.rb) and use https://www.npmjs.com/package/basedpyright

**Comment by nikaro** [comment]
@branchvincent this is what i initially tried but it didn't worked:
```
➤ basedpyright --version
node:internal/modules/cjs/loader:1251
  throw err;
  ^

Error: Cannot find module './dist/pyright'
Require stack:
- /opt/homebrew/Cellar/basedpyright/1.15.2/libexec/bin/basedpyright
    at Module._resolveFilename (node:internal/modules/cjs/loader:1248:15)
    at Module._load (node:internal/modules/cjs/loader:1074:27)
    at TracingChannel.traceSync (node:diagnostics_channel:315:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:217:24)
    at Module.require (node:internal/modules/cjs/loader:1339:12)
    at require (node:internal/modules/helpers:125:16)
    at Object.<anonymous> (/opt/homebrew/Cellar/basedpyright/1.15.2/libexec/bin/basedpyright:8:1)
    at Module._compile (node:internal/modules/cjs/loader:1546:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1691:10)
    at Module.load (node:internal/modules/cjs/loader:1317:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/opt/homebrew/Cellar/basedpyright/1.15.2/libexec/bin/basedpyright'
  ]
}

Node.js v22.6.0
```
It looks like the Python build pipeline does some more magic :-/

**Comment by nikaro** [comment]
Ah, looks like i messed up with links/symlinks:
```
nicolas@anubis:~/D/g/n/homebrew-tap|main⚡*
➤ /opt/homebrew/Cellar/basedpyright/1.15.2/libexec/bin/pyright --version
basedpyright 1.15.2
based on pyright 1.15.2
```

Here is the Formula i'm trying:
```
require "language/node"

class Basedpyright < Formula
  desc "Pyright fork with various improvements and built-in pylance features"
  homepage "https://github.com/DetachHead/basedpyright"
  url "https://registry.npmjs.org/basedpyright/-/basedpyright-1.15.2.tgz"
  sha256 "c9e0dcceb4c09c8a5426eb544c1c1991fd27415f313c12e594e1c3373bea6aa4"
  license "MIT"
  head "https://github.com/detachhead/basedpyright.git", branch: "main"

  depends_on "node"

  def install
    system "npm", "install", *std_npm_args
    ln "#{libexec}/bin/pyright", "#{libexec}/bin/basedpyright"
    ln "#{libexec}/bin/pyright-langserver", "#{libexec}/bin/basedpyright-langserver"
    bin.install_symlink Dir["#{libexec}/bin/basedpyright*"]
  end

  test do
    (testpath/"broken.py").write <<~EOS
      def wrong_types(a: int, b: int) -> str:
          return a + b
    EOS
    output = pipe_output("#{bin}/basedpyright broken.py 2>&1")
    assert_match "error: Expression of type \"int\" is incompatible with return type \"str\"", output
  end
end
```

**Comment by branchvincent** [comment]
I think we just want

```ruby
def install
  system "npm", "install", *std_npm_args
  bin.install_symlink libexec/"bin/pyright" => "basedpyright"
  bin.install_symlink libexec/"bin/pyright-langserver" => "basedpyright-langserver"
end
```

File: Formula/b/basedpyright.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+require "language/node"
+
```
Comment (branchvincent) [inline Formula/b/basedpyright.rb:2]: This is no longer needed
```suggestion
```

File: Formula/b/basedpyright.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    bin.install_symlink libexec/"bin/pyright-langserver" => "basedpyright-langserver"
+  end
+
+  test do
+    (testpath/"broken.py").write <<~EOS
+      def wrong_types(a: int, b: int) -> str:
+          return a + b
+    EOS
+    output = pipe_output("#{bin}/basedpyright broken.py 2>&1")
+    assert_match "error: Expression of type \"int\" is incompatible with return type \"str\"", output
```
Comment (branchvincent) [inline Formula/b/basedpyright.rb:25]: minor nit (ik this was actually from `pyright`)
```suggestion
    assert_match 'error: Expression of type "int" is incompatible with return type "str"', output
```

File: Formula/b/basedpyright.rb (hunk 1 above)
Comment (nikaro) [reply Formula/b/basedpyright.rb]: Good to know! Thanks.

**Review by branchv — APPROVED** [review]
Looks good, thanks @nikaro!

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@nikaro, thanks for your contribution to Homebrew! :tada: 🥇

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request merged**